docs/DEVELOPING.md

```diff
@@ -1,34 +1,103 @@
+
+After deployment:
+- SSH into AmpliPi (`ssh pi@{IP}`).
+- Navigate to the development root: `cd ~/amplipi-dev`.
+- Run the server in debug mode: `./scripts/run_debug_webserver`, which will start a webserver at [amplipi.local:5000](http://amplipi.local:5000).
+- Restart the AmpliPi service when done with `systemctl --user restart amplipi`.
+
+### Developing on an AmpliPi Controller
+
+To directly develop on an AmpliPi controller over SSH:
+
+1. Clone the repository on the AmpliPi at `~/amplipi-dev`
+2. Make your changes using your preferred editor.
+3. Run the server in debug mode: `./scripts/run_debug_webserver`
+4. When ready, run `./scripts/configure.py --python-deps --os-deps --display --web` to install necessary dependencies and reconfigure services.
+
+---
+
+## Mocked Development Setups
+
+If you don’t have an AmpliPi system, you can use a mocked setup for development and testing.
+
+### Mocked Audio & Controller (Debian-Based System)
+
+This method allows development on systems like a Raspberry Pi or Ubuntu.
+
+Supports:
+- Web interface development
+- API testing
+- Basic stream testing
+
+To set it up:
+1. Clone the repository.
+2. Install streaming dependencies using `./scripts/configure.py --os-deps`.
+3. Install Python dependencies with `./scripts/configure.py --python-deps`.
+4. Run the mock server with `./scripts/run_debug_webserver` (add the `--mock-streams` flag if you didn’t install the streaming dependencies).
+
+### Mocked Controller with 4 Stereo Audio Channels (Raspberry Pi OS)
 
-## Developing the webapp
-  To develop for the webapp, you will need to install nodejs and any dependencies for [ReactJS](https://react.dev/)
+This method requires a Raspberry Pi with a compatible USB audio device.
 
-## Developing on a separate computer
+Supports:
+- Web interface development
+- API testing
+- Streams integration and testing
 
-This allows remote development with the ability to test changes on your AmpliPi
-1. Checkout [this repo](https://github.com/micro-nova) on a linux based system. A git bash shell on windows works fine as well, we suggest Windows Subsystem for Linux (WSL).
-2. Make changes with your favorite editor, we suggest vscode
-3. Use `scripts/deploy` to deploy the latest software.
-   The pi must have access to the internet to successfully run this script.
+Setup:
+1. Install a 32-bit version of Raspberry Pi OS (older than December 2020).
+2. Connect a compatible 7.1 channel USB audio device.
+3. Clone the repository and modify the `asound.conf` to configure the audio device.
+4. Deploy using `scripts/deploy USER@HOSTNAME --mock-ctrl`.
+5. Run the mock controller with `./scripts/run_debug_webserver --mock-ctrl`.
 
-### Installing Node.js
+---
+## Developing the Frontend
+Our frontend is a React webapp, to develop it you must first download and install a few dependencies:
+- [React](https://react.dev/learn/installation)
```

Comment:
you don't need to "download and install react"

Reply:
Not technically a requirement I suppose but I would heavily suggest installing react dev tools so you can compile and test the frontend
Though perhaps I've forgotten the proper process now that I'm 3 years past setting up my React environment for the first and only time